src/index.js

```diff
@@ -1 +1,88 @@
-// Lets do this!
+import { toRefs, h, computed, markRaw, watch, getCurrentInstance } from 'vue'
+import { useForm, useField } from 'vee-validate';
+
+/**
+ * For a Schema, find the elements in each of the rows and remap the element with the given function
+ * @param {Array} schema
+ * @param {Function} fn
+ *
+ * @returns {Array}
+ */
+export const mapElementsInSchema = (schema, fn) => schema.map(row => row.map(el => fn(el)))
+
+export default function VeeValidatePluginFactory () {
+  return function VeeValidatePlugin(baseReturns) {
+    // Take the parsed schema from SchemaForm setup returns
+    const { parsedSchema, formBinds } = baseReturns
+  
+  
+    // Get additional properties not defined on the `SchemaForm` derivatives
+    const { attrs: formAttrs } = getCurrentInstance();
+    // Create a form context and inject the validation schema if needed
+    const { handleSubmit } = useForm({
+      validationSchema: formAttrs['validation-schema'] || formAttrs['validationSchema']
+    });
+    // Map components in schema to enhanced versions with `useField`
+    const formSchema = mapElementsInSchema(parsedSchema.value, el => {
+      return {
+        ...el,
+        component: markRaw(withField(el.component))
+      }
+    })
+
+    const boundSubmit = formBinds.value.onSubmit;  
+    const onSubmit = handleSubmit((_, { evt }) => {
+      boundSubmit(evt);
+    });
+  
+    return {
+      ...baseReturns,
+      formBinds: computed(() => {
+        return {
+          ...baseReturns.formBinds.value,
+          onSubmit,
+        }
+      }),
+      parsedSchema: computed(() => formSchema)
+    }
+  }
+} 
+
+export function withField (Comp) {
+  return {
+    name: 'withFieldWrapper',
+    props: {
+      modelValue: {
+        type: [String, Number],
+        default: undefined,
+      },
+      validations: {
+        type: [String, Object, Function],
+        default: undefined,
+      }
+    },
+    setup (props, { attrs }) {
+      const { validations, modelValue, model } = toRefs(props)
+      const fieldName = model ? model.value : attrs.model
+      const initialValue = modelValue ? modelValue.value : undefined
+      const { value, errorMessage } = useField(fieldName, validations || attrs.rules, {
+        initialValue
+      })
```

Comment:
Again, this may come out of my ignorance of your API, but if you pass down the `initialValue` like this, you're not breaking reactivity? Since `modelValue.value` will unwrap the value?

Reply:
No worries. vee-validate automatically unwraps reactive initial values and assigns their inner value to the output `value` binding, this is to avoid mutating props that people might have passed with `toRef` or `computed`.

So right there, I'm starting off with the initial value in the model. But later I'm making sure the `value` and `modelValue` are in sync with a watcher:

https://github.com/formvuelate/formvuelate-plugin-vee-validate/blob/feat/initial-implementation/src/index.js#L72-L76

Since vee-validate has no way to change the value as a formvuelate plugin, this one-way binding is fine I think